src/python.cc

```diff
@@ -798,6 +801,35 @@ ModelInstanceState::SetupChildProcess()
             .c_str());
   }
 
+  // Path to the extracted Python env
+  std::string python_execution_env = "none";
+
+  // Path to environment activation script
+  std::string path_to_activate;
+  std::string path_to_libpython;
+
+  if (model_state->PythonExecutionEnv() != "") {
+    try {
+      python_execution_env =
```

Comment:
We shouldn't need a copy of the execution environment for each instance, or even each model. If multiple models/instances use the same execution environment they should all use the same extracted version of the env. So it seems like the execution environments should be managed at the backend level.

Reply:
@deadeyegoodwin We do not copy the execution environment for each instance. The `Extract` function will extract the environment only if it does not exist in the `env_map_`. Otherwise, it will return the path to the previously extracted environment. Perhaps I should rename the `Extract` function to `ExtractIfNotExists` so that it is more clear.

Reply:
Improved the comments and function name to address this.

Reply:
ExtractIfNotExists feels like it could be misinterpreted. How about ExtractIfNotExtracted or similar?

Reply:
Sounds good, I'll use `ExtractIfNotExtracted `.